### PR TITLE
Modify metric api as per offline feebacks

### PR DESCRIPTION
**Pull request by cijothomas**

### CODE REVIEW

**Comment by cijothomas** [comment]
@tarekgh

File: src/OpenTelemetry.Api/Metrics/Meter.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        /// Creates a counter for long with given name.
+        /// </summary>
+        /// <param name="name">The name of the counter.</param>
+        /// <param name="monotonic">indicates if only positive values are expected.</param>
+        /// <returns>The counter instance.</returns>
+        public abstract Counter<long> CreateLongCounter(string name, bool monotonic = true);
+
+        /// <summary>
+        /// Creates a counter for double with given name.
+        /// </summary>
```
Comment (tarekgh) [inline src/OpenTelemetry.Api/Metrics/Meter.cs]: This looks missing CreateXXXXMeasure methods?

File: src/OpenTelemetry.Api/Metrics/Meter.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        /// <param name="monotonic">indicates if only positive values are expected.</param>
+        /// <returns>The guage instance.</returns>
+        public abstract Guage<double> CreateDoubleGauge(string name, bool monotonic = false);
+
+        /// <summary>
+        /// Constructs or retrieves the <see cref="LabelSet"/> from the given label key-value pairs.
+        /// </summary>
+        /// <param name="labels">Label key value pairs.</param>
+        /// <returns>The <see cref="LabelSet"/> with given label key value pairs.</returns>
+        public abstract LabelSet GetLabelSet(IEnumerable<KeyValuePair<string, string>> labels);
```
Comment (tarekgh) [inline src/OpenTelemetry.Api/Metrics/Meter.cs]: do we need to add the overloads which take KeyValuePair parameters?

**Comment by tarekgh** [comment]
Add a couple of minor comments. LGTM otherwise. Thanks for having this ready.

File: src/OpenTelemetry.Api/Metrics/Meter.cs (hunk 2 above)
Comment (cijothomas) [reply src/OpenTelemetry.Api/Metrics/Meter.cs]: yes will do in a follow up pr.

**Comment by cijothomas** [comment]
@SergeyKanzhelev Please merge.

**Pull request merged**